src/lib/tcp/src/connection.rs

```diff
@@ -240,7 +240,16 @@ impl<I: Instant> Connection<I> {
                 #[cfg(debug_assertions)]
                 debug_assert!(wants_to_send);
 
-                break 'packet (self.send.buffer.next_seq(), TcpFlags::empty(), Bytes::new());
+                // use the sequence number of the next unsent message if we have one buffered,
```

Comment:
Optional - maybe it's worth writing additional unit/regression tests that'd catch this?

Reply:
Added a unit test that tries to cover this.

Reply:
I also had to disable miri for this new test, otherwise it times-out.